source/common/common/assert.h

```diff
@@ -1,8 +1,54 @@
 #pragma once
 
+#include <functional>
+
 #include "common/common/logger.h"
 
 namespace Envoy {
+namespace Assert {
+
+class ActionRegistration {
+public:
+  virtual ~ActionRegistration() {}
+};
+typedef std::unique_ptr<ActionRegistration> ActionRegistrationPtr;
+
+/**
+ * Sets an action to be invoked when a debug assertion failure is detected
+ * in a release build. This action will be invoked each time an assertion
+ * failure is detected.
+ *
+ * This function is not thread-safe; concurrent calls are not allowed.
```

Comment:
FYI there is no guarantee that asserts will not fire concurrently in the current code. I'm not sure it matters with what you are doing in your invoked function, but do you want to do something about this or change the comment here?

Reply:
I'll clarify the comment; what I mean here is that it isn't safe to concurrently set the assert callback, not that the assert callback cannot be called concurrently.